pkg/execution/executor/executor.go

```diff
@@ -551,22 +527,84 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 	// contains it.
 	md := s.Metadata()
 
+	// Retrieve the start time of the function
+	fnStartTime := time.Now()
+	if md.GetStartTime() != nil {
+		fnStartTime = *md.GetStartTime()
+	}
+
 	// Store the metadata in context for future use and propagate trace
 	// context. This can be used to reduce reads in the future.
 	ctx = e.extractTraceCtx(WithContextMetadata(ctx, md), id, &item)
```

Comment:
We already inject and then extract the tracing context from either the run's metadata or the queue item here.

Why do we now also add a separate `SpanID`?

Reply:
We need to reconstruct to make it a duplicate. 👍